### PR TITLE
Switch to org.hibernate.testing.orm.junit.RequiresDialect to use base Dialect classes instead of version specific ones

### PR DESCRIPTION
**Pull request by jrenaat**

### CODE REVIEW

File: hibernate-core/src/test/java/org/hibernate/orm/test/jpa/query/NativeQueryOrdinalParametersTest.java [hunk 1]
```diff
@@ -18,106 +18,114 @@
...
+		);
 	}
 
 	@Test
 	@TestForIssue(jiraKey = "HHH-12532")
 	// Add RequiresDialect be Cockroach version 201
-	@RequiresDialect({ PostgreSQL82Dialect.class})
-	public void testCteNativeQueryOrdinalParameter() {
+	@RequiresDialect( value = PostgreSQLDialect.class, version = 820 )
+	@RequiresDialect( value = CockroachDialect.class, version = 820 )
```
Comment (beikov) [inline hibernate-core/src/test/java/org/hibernate/orm/test/jpa/query/NativeQueryOrdinalParametersTest.java:127]: I think this should be this way:
```suggestion
	@RequiresDialect( value = CockroachDialect.class, version = 2010 )
```

**Review by beikov — COMMENTED** [review]
Looking good mostly, but there is a compilation error:

```
2021-11-09T18:38:33.6414987Z /home/runner/work/hibernate-orm/hibernate-orm/hibernate-testing/src/main/java/org/hibernate/testing/orm/junit/EntityManagerFactoryExtension.java:179: error: incompatible types: Class<CAP#1> cannot be converted to Class<? extends Provider<?>>
2021-11-09T18:38:33.6416296Z 
2021-11-09T18:38:33.6417047Z > Task :hibernate-testing:compileJava
2021-11-09T18:38:33.6418672Z 			final Class<? extends SettingProvider.Provider<?>> providerImpl = providerAnn.provider();
```

File: hibernate-testing/src/main/java/org/hibernate/testing/orm/junit/EntityManagerFactoryExtension.java [hunk 2]
```diff
@@ -176,16 +175,14 @@ public static EntityManagerFactoryScope findEntityManagerFactoryScope(
...
-				final Class<? extends NonStringValueSettingProvider> _class = emfAnn.nonStringValueSettingProviders()[i];
-				try {
-					NonStringValueSettingProvider valueProvider = _class.newInstance();
-					integrationSettings.put( valueProvider.getKey(), valueProvider.getValue() );
-				}
-				catch (Exception e) {
-					log.error( "Error obtaining special value for " + _class.getName(), e );
-				}
+		for ( SettingProvider providerAnn : emfAnn.settingProviders() ) {
+			final Class<? extends SettingProvider.Provider<?>> providerImpl = providerAnn.provider();
```
Comment (dreab8) [inline hibernate-testing/src/main/java/org/hibernate/testing/orm/junit/EntityManagerFactoryExtension.java:179]: I think this should be `final Class<? extends SettingProvider.Provider> providerImpl = providerAnn.provider();`

File: hibernate-core/src/test/java/org/hibernate/orm/test/dialect/functional/MariaDBExtractSequenceMetadataTest.java [hunk 3]
```diff
@@ -51,24 +51,26 @@ public static void setUpDBs() throws Exception {
...
 
-	@AfterClass
+	@AfterAll
 	public static void tearDownDBs() throws SQLException {
 		try (Connection conn = getConnection()) {
 			try (Statement stmt = conn.createStatement()) {
 				stmt.execute( "DROP DATABASE IF EXISTS " + secondaryDbName );
 			}
 			catch (Exception e) {
+				int i = 0;
```
[the remark this answers is not in the document]
Comment (jrenaat) [reply hibernate-core/src/test/java/org/hibernate/orm/test/dialect/functional/MariaDBExtractSequenceMetadataTest.java:65]: Yes, clearly :-)

**Review by beikov — APPROVED** [review]

**Comment by beikov** [comment]
There are some test failures: https://ci.hibernate.org/blue/organizations/jenkins/hibernate-orm-pipeline/detail/PR-4322/18/tests

**Pull request merged**